File: frontend/src/types/contracts/auth.contract.ts
/**
 * 认证相关类型契约
 * 与backend/app/schemas/contracts/auth_contract.py对应
 */

export interface JWTPayload {
  sub: string; // user_id
  tenant: string; // tenant_id
  exp: number;
  iat: number;
  permissions: string[];
}

export interface UserContext {
  user_id: string;
  tenant_id: string;
  permissions: string[];
  session_id: string;
  expires_at: string; // ISO string
}

// 前端认证状态管理
export interface AuthState {
  isAuthenticated: boolean;
  user: UserContext | null;
  token: string | null;
  refreshToken: string | null;
  loading: boolean;
  error: string | null;
}

export interface LoginRequest {
  username: string;
  password: string;
}

export interface LoginResponse {
  access_token: string;
  refresh_token: string;
  token_type: string;
  expires_in: number;
  user: UserContext;
}

// 认证相关组件Props
export interface ProtectedRouteProps {
  children: React.ReactNode;
  requiredPermissions?: string[];
  fallback?: React.ReactNode;
}

export interface AuthContextProps {
  authState: AuthState;
  login: (credentials: LoginRequest) => Promise<void>;
  logout: () => void;
  refreshTokens: () => Promise<void>;
  hasPermission: (permission: string) => boolean;
}

File: backend/app/schemas/contracts/auth_contract.py
from __future__ import annotations

from datetime import datetime
from typing import List

from pydantic import BaseModel, Field


class JWTPayload(BaseModel):
    sub: str = Field(..., description="user_id")
    tenant: str = Field(..., description="tenant_id")
    exp: int
    iat: int
    permissions: List[str] = Field(default_factory=list)


class UserContext(BaseModel):
    user_id: str
    tenant_id: str
    permissions: List[str] = Field(default_factory=list)
    session_id: str
    expires_at: datetime
